pkg/tso/tso.go

```diff
@@ -74,6 +76,65 @@ type timestampOracle struct {
 	lastSavedTime atomic.Value // stored as time.Time
 	suffix        int
 	dcLocation    string
+
+	// pre-initialized metrics
+	// timestampOracle event counter
+	syncEvent             prometheus.Counter
```

Comment:
How about putting it all into an independent structure? then like `t.metrics.syncEvent.Inc()`